src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -212,13 +212,28 @@ public static BicepCompletionContext Create(IFeatureProvider featureProvider, Co
                 // check if we're inside an expression
                 kind |= ConvertFlag(IsInnerExpressionContext(matchingNodes, offset), BicepCompletionContextKind.Expression);
 
-                if (kind.HasFlag(BicepCompletionContextKind.Expression) &&
-                    PropertyTypeShouldFlowThrough(matchingNodes, propertyInfo, offset))
+                if (kind.HasFlag(BicepCompletionContextKind.Expression))
                 {
-                    kind |= BicepCompletionContextKind.PropertyValue;
+                    if (PropertyTypeShouldFlowThrough(matchingNodes, propertyInfo, offset))
+                    {
+                        kind |= BicepCompletionContextKind.PropertyValue;
+                    }
+
+                    var arrayItemInfo = SyntaxMatcher.FindLastNodeOfType<ArrayItemSyntax, ArrayItemSyntax>(matchingNodes);
+                    if (ArrayItemTypeShouldFlowThrough(matchingNodes, arrayItemInfo))
+                    {
+                        kind |= BicepCompletionContextKind.ArrayItem;
+                    }
                 }
             }
 
+            if (propertyInfo.node?.TryGetKeyText() == LanguageConstants.ResourceDependsOnPropertyName
+                && topLevelDeclarationInfo.node is ResourceDeclarationSyntax
```

Comment:
We'll also need to check for `ModuleDeclarationSyntax`, and ensure you've got a test to cover e.g.
```bicep
module foo 'foo.bicep' = {
  dependsOn: [ | ]
}
```

Reply:
I'm assuming modules can depend on both symbol types. Is that correct?